AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/nativead/ManualNativeLateBindingAdActivity.java

```diff
@@ -0,0 +1,130 @@
+import android.view.View;
+import android.widget.Button;
+import android.widget.FrameLayout;
+
+import com.adjust.sdk.Adjust;
+import com.adjust.sdk.AdjustAdRevenue;
+import com.adjust.sdk.AdjustConfig;
+import com.applovin.enterprise.apps.demoapp.R;
+import com.applovin.enterprise.apps.demoapp.ui.BaseAdActivity;
+import com.applovin.mediation.MaxAd;
+import com.applovin.mediation.MaxError;
+import com.applovin.mediation.nativeAds.MaxNativeAdListener;
+import com.applovin.mediation.nativeAds.MaxNativeAdLoader;
+import com.applovin.mediation.nativeAds.MaxNativeAdView;
+import com.applovin.mediation.nativeAds.MaxNativeAdViewBinder;
+
+public class ManualNativeLateBindingAdActivity
+        extends BaseAdActivity
+{
+    private MaxNativeAdLoader nativeAdLoader;
+    private FrameLayout       nativeAdLayout;
+    private Button            showAdButton;
+
+    private MaxAd nativeAd;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate( savedInstanceState );
+        setContentView( R.layout.activity_native_manual_late_binding );
+        setTitle( R.string.activity_manual_native_late_binding_ad );
+
+        nativeAdLayout = findViewById( R.id.native_ad_layout );
+        showAdButton = findViewById( R.id.show_ad_button );
+        setupCallbacksRecyclerView();
+
+        nativeAdLoader = new MaxNativeAdLoader( "YOUR_AD_UNIT_ID", this );
+        nativeAdLoader.setRevenueListener( ad -> {
+            logAnonymousCallback();
+
+            AdjustAdRevenue adjustAdRevenue = new AdjustAdRevenue( AdjustConfig.AD_REVENUE_APPLOVIN_MAX );
+            adjustAdRevenue.setRevenue( ad.getRevenue(), "USD" );
+            adjustAdRevenue.setAdRevenueNetwork( ad.getNetworkName() );
+            adjustAdRevenue.setAdRevenueUnit( ad.getAdUnitId() );
+            adjustAdRevenue.setAdRevenuePlacement( ad.getPlacement() );
+
+            Adjust.trackAdRevenue( adjustAdRevenue );
+        } );
+        nativeAdLoader.setNativeAdListener( new MaxNativeAdListener()
+        {
+            @Override
+            public void onNativeAdLoaded(final MaxNativeAdView nativeAdView, final MaxAd ad)
+            {
+                logAnonymousCallback();
+
+                // Cleanup any pre-existing native ad to prevent memory leaks.
+                if ( nativeAd != null )
+                {
+                    nativeAdLoader.destroy( nativeAd );
+                }
+
+                // Save ad to be rendered later.
+                nativeAd = ad;
+
+                showAdButton.setEnabled( true );
+            }
+
+            @Override
+            public void onNativeAdLoadFailed(final String adUnitId, final MaxError error)
+            {
+                logAnonymousCallback();
+            }
+
+            @Override
+            public void onNativeAdClicked(final MaxAd ad)
+            {
+                logAnonymousCallback();
+            }
+        } );
+    }
+
+    @Override
+    protected void onDestroy()
```

Comment:
nit: should move this to the bottom of the file.

Reply:
I think we do this in other Activities so might be fine for now and we can just cleanup everything later